test/extended/networking/ipsec.go

```diff
@@ -0,0 +1,504 @@
+	tcpdumpESPFilter = "esp or udp port 4500 and src %s and dst %s"
+	// tcpdumpGeneveFilter can be used to filter out Geneve encapsulated packets destined to target node.
+	tcpdumpGeneveFilter          = "udp port 6081 and src %s and dst %s"
+	masterIPsecMachineConfigName = "80-ipsec-master-extensions"
+	workerIPSecMachineConfigName = "80-ipsec-worker-extensions"
+)
+
+// configureIPsec helps to rollout specified IPsecConfig on the cluster.
+func configureIPsec(oc *exutil.CLI, ipsecConfig *v1.IPsecConfig) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		network, err := oc.AdminOperatorClient().OperatorV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		network.Spec.DefaultNetwork.OVNKubernetesConfig.IPsecConfig = ipsecConfig
+		_, err = oc.AdminOperatorClient().OperatorV1().Networks().Update(context.Background(), network, metav1.UpdateOptions{})
+		return err
+	})
+}
+
+// configureIPsec helps to rollout specified IPsec Mode on the cluster. If the cluster is already configured with specified mode, then
+// this is almost like no-op for the cluster.
+func configureIPsecMode(oc *exutil.CLI, ipsecMode v1.IPsecMode) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		network, err := oc.AdminOperatorClient().OperatorV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if network.Spec.DefaultNetwork.OVNKubernetesConfig.IPsecConfig == nil {
+			network.Spec.DefaultNetwork.OVNKubernetesConfig.IPsecConfig = &v1.IPsecConfig{Mode: ipsecMode}
+		} else if network.Spec.DefaultNetwork.OVNKubernetesConfig.IPsecConfig.Mode != ipsecMode {
+			network.Spec.DefaultNetwork.OVNKubernetesConfig.IPsecConfig.Mode = ipsecMode
+		} else {
+			// No changes to existing mode, return without updating networks.
+			return nil
+		}
+		_, err = oc.AdminOperatorClient().OperatorV1().Networks().Update(context.Background(), network, metav1.UpdateOptions{})
+		return err
+	})
+}
+
+func getIPsecConfig(oc *exutil.CLI) (*v1.IPsecConfig, error) {
+	network, err := oc.AdminOperatorClient().OperatorV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	ipsecConfig := network.Spec.DefaultNetwork.OVNKubernetesConfig.IPsecConfig
+	if ipsecConfig != nil {
+		ipsecConfig = ipsecConfig.DeepCopy()
+	}
+	return ipsecConfig, nil
+}
+
+// ensureIPsecEnabled this function ensure IPsec is enabled by making sure ovn-ipsec-host daemonset
+// is completely ready on the cluster and cluster operators are coming back into ready state
+// once ipsec rollout is complete.
+func ensureIPsecEnabled(oc *exutil.CLI) (bool, error) {
+	mcComplete, err := ensureIPsecMachineConfigRolloutComplete(oc)
+	if !mcComplete {
+		return mcComplete, err
+	}
+	for {
+		running, err := isIPsecDaemonSetRunning(oc)
+		if err != nil && !isConnReset(err) {
```

Comment:
yes, saw this error on many attempts and also experienced the same while writing tests for ExternalIP earlier.
https://github.com/openshift/origin/blob/master/test/extended/networking/services.go#L147

Reply:
OK. If we had an overall reasonable timeout, we could potentially just ignore any error.

What I am afraid here is that we get a different type of error if a request was being processed while the kapi service was being shutdown (expectedly). I guess in theory that should not happen as the service would try to service all request before shutting down, but I consider that best effort rather than a guaranteed thing that will happen.